modules/web/src/app/core/services/rbac.ts

```diff
@@ -14,33 +14,76 @@
 
 import {HttpClient, HttpHeaders} from '@angular/common/http';
 import {Injectable} from '@angular/core';
-import {Observable} from 'rxjs';
+import {Observable, timer, startWith, Subject, merge} from 'rxjs';
 
 import {environment} from '@environments/environment';
+import _ from 'lodash';
+import {map, switchMap, tap, distinctUntilChanged} from 'rxjs/operators';
 import {
   Binding,
   ClusterBinding,
+  ClusterNamespace,
   ClusterRoleName,
   CreateBinding,
   DeleteBindingBody,
   Kind,
   RoleName,
 } from '@shared/entity/rbac';
+import {AppConfigService} from '@app/config.service';
 
 @Injectable()
 export class RBACService {
+  private readonly _refreshTime = 10;
   private _newRestRoot: string = environment.newRestRoot;
+  private _refreshTimer$ = timer(0, this._appConfigService.getRefreshTimeBase() * this._refreshTime);
 
-  constructor(private readonly _http: HttpClient) {}
+  private _refreshClusterBindings$ = new Subject<void>();
+  private _refreshNamespaceBindings$ = new Subject<void>();
+
+  private _clusterNamespacesMap = new Map<string, string[]>();
+  private _namespaceRoleNameMap = new Map<string, RoleName[]>();
+  private _clusterRoleNamesMap = new Map<string, ClusterRoleName[]>();
+  private _clusterBindingMap = new Map<string, ClusterBinding[]>();
+  private _namespaceBindingsMap = new Map<string, Binding[]>();
+
+  constructor(private readonly _http: HttpClient, private readonly _appConfigService: AppConfigService) {}
 
   getClusterRoleNames(clusterID: string, projectID: string): Observable<ClusterRoleName[]> {
```

Comment:
We should use the same approach (with `shareReplay`) as `ClusterServiceAccountService.get` method in all these methods here.